phd-tests/framework/src/artifacts/mod.rs

```diff
@@ -62,44 +59,9 @@ struct Artifact {
     untar: Option<camino::Utf8PathBuf>,
 }
 
-impl FromStr for Commit {
-    type Err = anyhow::Error;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let s = s.trim();
-
-        // Ensure this looks like a valid Git commit.
-        anyhow::ensure!(
-            s.len() == 40,
-            "Buildomat requires full (40-character) Git commit hashes"
-        );
-
-        for c in s.chars() {
-            if !c.is_ascii_hexdigit() {
-                anyhow::bail!(
-                    "'{c}' is not a valid hexadecimal digit; Git \
-                    commit hashes should consist of the characters \
-                    [0-9, a-f, A-F]"
-                );
-            }
-        }
-
-        Ok(Self(s.to_string()))
-    }
-}
-
-impl std::fmt::Display for Commit {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.write_str(&self.0)
-    }
-}
-
-impl<'de> Deserialize<'de> for Commit {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
-        let s = String::deserialize(deserializer)?;
-        FromStr::from_str(&s).map_err(serde::de::Error::custom)
-    }
+#[derive(Debug)]
+struct DownloadConfig {
+    timeout: Duration,
+    buildomat_backoff: backoff::ExponentialBackoff,
```

Comment:
naming nit: I would rename the `buildomat`-flavored members and parameters in this change (e.g. also `max_buildomat_wait` where it appears) to something a bit more generic, since IIUC this will also apply to guest OS image downloads, which I expect will come from other places (S3 for now or, in the hopefully-not-too-distant future, catacomb)

Reply:
This only applies to downloads from Buildomat; there are separate functions for downloading from Buildomat sources and downloading from the list of remote URIs, and the retry logic is only used for Buildomat downloads. 

I'm happy to also add retries to downloads from `Remote` artifact sources, but I think they ought to be configured separately, since we default to a _very_ long max backoff duration for Buildomat as a way of waiting for an artifact to be published. When downloading from S3, I think we probably don't want to default to retrying failed downloads for 20 minutes...

Reply:
Oh, right, I forgot what the motivation was and then missed this detail when I skimmed the download logic. Alright, makes sense to me!

Reply:
i'll add some comments explaining this so that it's hopefully obvious to future generations :)